src/main/scala/org/labrad/Server.scala

```diff
@@ -262,13 +264,58 @@ object Server {
    * Includes basic parsing of command line options.
    */
   def run(s: Server[_, _], args: Array[String]): Unit = {
-    val options = Util.parseArgs(args, Seq("name", "host", "port", "password"))
+    val parser = new ArgotParser("labrad",
+      preUsage = Some("The labrad manager."),
+      sortUsage = false
+    )
+    val nameOpt = parser.option[String](
+      names = List("name"),
+      valueName = "string",
+      description = "The name to use for this server when connecting to the " +
+        "labrad manager. If given, this will override the default name. " +
+        "This is generally not needed, but can be useful in testing if you " +
+        "want to run several copies of a given server without name conflicts."
+    )
+    val hostOpt = parser.option[String](
+      names = List("host"),
+      valueName = "string",
+      description = "The hostname where the labrad manager is running. " +
+        "If not provided, fallback to the value given in the LABRADHOST " +
+        "environment variable, with default value 'localhost'."
+    )
+    val portOpt = parser.option[Int](
+      names = List("port"),
+      valueName = "int",
+      description = "Port on which to connect to the labrad manager. " +
+        "If not provided, fallback to the value given in the LABRADPORT " +
+        "environment variable, with default value 7682."
+    )
+    val passwordOpt = parser.option[String](
+      names = List("password"),
+      valueName = "string",
+      description = "Password to use to authenticate to the manager. " +
+        "If not provided, fallback to the value given in the LABRADPASSWORD " +
+        "environment variable, with default value '' (empty password)."
+    )
+    val help = parser.flag[Boolean](List("h", "help"),
+      "Print usage information and exit")
+
+    try {
+      parser.parse(args)
```

Comment:
You could add a wrapper function here to remove "=" signs and so allow usage of the old or new parameter setting method.